tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -0,0 +1,605 @@
+    public void SyncDevDrives()
+    {
+        // temporary, we'll need to give the users a way to disable this.
+        // if they don't want to use hyper-v
+        _toastNotificationService.CheckIfUserIsAHyperVAdmin();
+        GetDevDrives();
+    }
+
+    /// <summary>
+    /// Make sure we only get the list of DevDrives from the DevDriveManager once when the page is first navigated to.
+    /// All other times will be through the use of the sync button.
+    /// </summary>
+    protected async override Task OnFirstNavigateToAsync()
+    {
+        // Do nothing, but we need to override this as the base expects a task to be returned.
+        await Task.CompletedTask;
+
+        // temporary, we'll need to give the users a way to disable this.
+        // if they don't want to use hyper-v
+        _toastNotificationService.CheckIfUserIsAHyperVAdmin();
+
+        GetDevDrives();
+        GetDevDriveOptimizers();
+        GetDevDriveOptimizeds();
+    }
+
+    public void UpdateNextButtonState()
+    {
+        // Only enable the next button if the DevDriveManager has a DevDrive selected to apply the configuration to.
+        // and if the DevDriveManager has finished loading the dev drives.
+        _setupFlowOrchestrator.NotifyNavigationCanExecuteChanged();
+        SyncDevDrivesCommand.NotifyCanExecuteChanged();
+    }
+
+    /// <summary>
+    /// Starts the process of getting the list of DevDriveOptimizers. the sync and next
+    /// buttons should be disabled when work is being done.
+    /// </summary>
+    private void GetDevDriveOptimizers()
+    {
+        // We need to run this on a background thread so we don't block the UI thread.
+        Task.Run(async () =>
+        {
+            await _dispatcher.EnqueueAsync(async () =>
+            {
+                // Remove any existing DevDriveOptimizersListViewModels from the list if they exist.
+                RemoveDevDriveOptimizersListViewModels();
+
+                // Disable the sync and next buttons while we're getting the dev drives.
+                DevDriveOptimizerLoadingCompleted = false;
+
+                // load the dev drives so we can show them in the UI.
+                await LoadAllDevDriveOptimizersInTheUI();
+
+                // Enable the sync and next buttons when we're done getting the dev drives.
+                UpdateNextButtonState();
+
+                DevDriveOptimizersCollectionView.Refresh();
+            });
+        });
+    }
+
+    /// <summary>
+    /// Removes all DevDriveOptimizersListViewModels from the list view model list and removes the dev drive
+    /// selected to apply the configuration to. This should refresh the UI to show no dev drives.
+    /// </summary>
+    private void RemoveDevDriveOptimizersListViewModels()
+    {
+        var totalLists = _devDriveOptimizerViewModelList.Count;
+        for (var i = totalLists - 1; i >= 0; i--)
+        {
+            _devDriveOptimizerViewModelList.RemoveAt(i);

```

Comment:
Removing during a loop is a bad idea.  Consider using [RemoveAll](https://learn.microsoft.com/dotnet/api/system.collections.generic.list-1.removeall?view=net-8.0)

Reply:
ObservableCollection does not have RemoveAll defined. Once I change to list from list of lists, I am guessing this should go away.